test/e2e/example/example.go

```diff
@@ -0,0 +1,135 @@
+/*
+Copyright 2019 The Submariner Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package example
+
+import (
+    "fmt"
+    "k8s.io/apimachinery/pkg/util/wait"
+    "time"
+
+    "github.com/rancher/submariner/test/e2e/framework"
+
+    v1 "k8s.io/api/core/v1"
+    metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+    . "github.com/onsi/ginkgo"
+    . "github.com/onsi/gomega"
+)
+
+var _ = Describe("[example] Basic example to demonstrate how to write tests using the framework", func() {
+    f := framework.NewDefaultFramework("basic-example")
+    It("Should be able to list existing nodes on the cluster", func() {
+
+        testListingNodes(f)
+    })
+    It("Should be able to create a pod using the provided client", func() {
+
+        testCreatingAPod(f)
+    })
+})
+
+
+func testListingNodes(f *framework.Framework) {
+    cs := f.ClientSet
+    nc := cs.CoreV1().Nodes()
+    By("Requesting node list from API")
+    nodes, err := nc.List(metav1.ListOptions{})
+    Expect(err).NotTo(HaveOccurred())
+
+    if len(nodes.Items) == 0 {
+        Expect(fmt.Errorf("no Nodes in the cluster")).NotTo(HaveOccurred())
+    }
```

Comment:
This looks a little unusual - it seems you want to add context to the failure output. How about:

    Expect(len(nodes.Items)).ToNot(BeZero(), "No Nodes in the cluster")

Reply:
That looks much better, this is copy&paste from kubernetes E2E examples.